File: DSA/Arshdeep-Kaur/Day 68/Day68_Question1.cpp
class Solution {
public:
    void postOrder(TreeNode* node, vector<int> &ans) {
        //base case 
        if (node == NULL) {
            return;
        }

        postOrder(node -> left, ans);
        postOrder(node -> right, ans);
        ans.push_back(node -> val);
    }

    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> ans;

        postOrder(root, ans);
        return ans;
    }
};

File: DSA/Manav-Rajput/Day68_question1.java
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        postorderHelper(root, result);
        return result;
    }
    
    private void postorderHelper(TreeNode node, List<Integer> result) {
        if (node == null) {
            return;
        }
        
        // Recurse on the left subtree
        postorderHelper(node.left, result);
        
        // Recurse on the right subtree
        postorderHelper(node.right, result);
        
        // Visit the root node
        result.add(node.val);
    }
}
